mypy_protobuf/main.py

```diff
@@ -969,21 +1053,36 @@ def write_grpc_services(
                 if self._write_comments(scl):
                     wl("")
                 self.write_grpc_methods(service, scl)
-            server = self._import("grpc", "Server")
-            aserver = self._import("grpc.aio", "Server")
+
             if service.options.deprecated:
                 self._write_deprecation_warning(
                     scl + [d.ServiceDescriptorProto.OPTIONS_FIELD_NUMBER] + [d.ServiceOptions.DEPRECATED_FIELD_NUMBER],
                     "This servicer has been marked as deprecated using proto service options.",
                 )
+
             wl(
                 "def add_{}Servicer_to_server(servicer: {}Servicer, server: {}) -> None: ...",
                 service.name,
                 service.name,
-                f"{self._import('typing', 'Union')}[{server}, {aserver}]",
+                self.make_server_type(),
             )
             wl("")
 
+    def make_server_type(self) -> str:
+        server = aserver = None
+        if self.grpc_type.supports_sync:
+            server = self._import("grpc", "Server")
+        if self.grpc_type.supports_async:
+            aserver = self._import("grpc.aio", "Server")
+
+        if server and aserver:
+            return f"{self._import('typing', 'Union')}[{server}, {aserver}]"
+        elif server:
+            return server
+        elif aserver:
+            return aserver
+        raise RuntimeError(f"Impossible, {self.grpc_type=}")  # pragma: no cover
```

Comment:
What is the purpose of the pragma comment here?

The logic feels a bit confusing here. What about:

```
server = self._import("grpc", "Server")
aserver = self._import("grpc.aio", "Server")
if self.grpc_type == BOTH:
  return f"{self._import('typing', 'Union')}[{server}, {aserver}]"
elif self.grpc_type == ASYNC:
  return aserver
elif self.grpc_type == SYNC:
  return server
else:
  raise RuntimeError(f"Impossible, {self.grpc_type=}")
```